File: client/Domain/Events/Events.cs
using System.Text.Json.Serialization;
namespace client.Domain.Events;
public abstract class GameEvent { }
public class GameStartedEvent : GameEvent
{
    [JsonPropertyName("category")]
    public required string Category { get; set; }
    
    [JsonPropertyName("difficulty")]
    public required string Difficulty { get; set; }
}

public class PlayerJoinedEvent : GameEvent
{
    [JsonPropertyName("username")]
    public required string Username { get; set; }
}

public class PlayerLeftEvent : GameEvent
{
    [JsonPropertyName("username")]
    public required string Username { get; set; }
}

public class RoomUpdatedEvent : GameEvent
{
    [JsonPropertyName("room_id")]
    public required string RoomID { get; set; }
}

public class RoomDeletedEvent : GameEvent
{
    [JsonPropertyName("room_id")]
    public required string RoomID { get; set; }
}

File: server/internal/events/events.go
package events

type GameStartedEventPayload struct {
	Category   string `json:"category"`   // Категория игры
	Difficulty string `json:"difficulty"` // Сложность игры
}

type PlayerJoinedEventPayload struct {
	Username string `json:"username"` // Имя игрока
}

type PlayerLeftEventPayload struct {
	Username string `json:"username"` // Имя игрока
}

type RoomHasBeenDeletedEventPayload struct {
	RoomId string `json:"room_id"`
}

type RoomHasBeenUpdatedEventPayload struct {
	RoomId string `json:"room_id"`
}
